examples/run/run.cpp

```diff
@@ -1053,11 +1074,44 @@ static int get_user_input(std::string & user_input, const std::string & user) {
     return 0;
 }
 
+// Reads a chat template file to be used
+static std::string read_chat_template_file(const std::string & chat_template_file) {
+    if(chat_template_file.empty()){
+        return "";
+    }
+
+    FILE* file = ggml_fopen(chat_template_file.c_str(), "r");
+        if (!file) {
+        std::cerr << "Error opening chat template file '" << chat_template_file << "': " << strerror(errno) << "\n";
+        return "";
+    }
+
+    fseek(file, 0, SEEK_END);
+    size_t size = ftell(file);
+    fseek(file, 0, SEEK_SET);
+
+    std::vector<unsigned char> data(size);
+    size_t                     read_size = fread(data.data(), 1, size, file);
+    fclose(file);
+    if (read_size != size) {
+        std::cerr << "Error reading chat template file '" << chat_template_file << "': " << strerror(errno) << "\n";
+        return "";
+    }
+    return std::string(data.begin(), data.end());
+}
+
 // Main chat loop function
-static int chat_loop(LlamaData & llama_data, const std::string & user, bool use_jinja) {
+static int chat_loop(LlamaData & llama_data, const std::string & user, const std::string & chat_template_file, bool use_jinja) {
```

Comment:
I think at some point we might as well just do:

```
static int chat_loop(LlamaData & llama_data, const Opt & opt)
```

and pass in the whole Opt struct/class as a reference, not necessary for this PR though.

Reply:
Sounds good to me. Would do that in a follow-up PR.